src/runner.ts

```diff
@@ -22,7 +22,15 @@ export class ActionRunner {
 
     this.logger.info(`Obtained config at ${configLocation}`);
 
-    return validate<ConfigurationFile>(config, schema, { message: "Configuration file is invalid" });
+    const configFile = validate<ConfigurationFile>(config, schema, { message: "Configuration file is invalid" });
+
+    const [result, error] = validateRegularExpressions(configFile, this.logger);
+    if (!result) {
+      this.logger.error(error);
+      throw new Error("Regular expression is invalid. Check the logs");
```

Comment:
Maybe instead of `Regular expression is invalid. Check the logs`, do 
```ts
throw new Error(`Regular expression is invalid: ${error.message}`);
```
?

We wouldn't expect anything aside from descriptive error there, right? (because, there's nothing going on except compiling the regex). 

Or, leave it like this, it's good enough already.

Reply:
Yeah, it makes sense. I don't need to capture the state, only validate that it works or fail.

It also makes it easier for testing